recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -36,22 +46,41 @@ def requirements(self):
             self.requires("libpcap/1.10.1")
 
     def validate(self):
+        if self.settings.compiler.cppstd:
+            # popen()/pclose() usage
+            check_min_cppstd(self, self._min_cppstd, gnu_extensions=not is_msvc(self))
         if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
             raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
 
+    def validate_build(self):
+        compiler_version = Version(self.settings.compiler.version)
+        if self.settings.compiler == "gcc" and compiler_version < "5.1":
+            raise ConanInvalidConfiguration("PcapPlusPlus requires GCC >= 5.1")
+
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
         tc.variables["PCAPPP_BUILD_TESTS"] = False
         tc.variables["PCAPPP_BUILD_EXAMPLES"] = False
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
```

Comment:
```suggestion
```
This should not be necessary - set automatically by Conan and the project is using a high enough CMake policy version to not be affected by CMP0077 issues either.